diff -Naurp ./configure.ac ./configure.ac
--- ./configure.ac	2011-03-29 16:52:13.000000000 -0500
+++ ./configure.ac	2011-03-29 16:54:43.000000000 -0500
@@ -646,6 +646,9 @@ AM_CONDITIONAL(HAS_at90pwm316, test "x$H
 CHECK_AVR_DEVICE(at90pwm216)
 AM_CONDITIONAL(HAS_at90pwm216, test "x$HAS_at90pwm216" = "xyes")
 
+CHECK_AVR_DEVICE(at90pwm161)
+AM_CONDITIONAL(HAS_at90pwm161, test "x$HAS_at90pwm161" = "xyes")
+
 CHECK_AVR_DEVICE(at94k)
 AM_CONDITIONAL(HAS_at94k, test "x$HAS_at94k" = "xyes")
 
@@ -1170,6 +1173,7 @@ AC_CONFIG_FILES([
 	avr/lib/avr5/at90can128/Makefile
 	avr/lib/avr5/at90pwm216/Makefile
 	avr/lib/avr5/at90pwm316/Makefile
+	avr/lib/avr5/at90pwm161/Makefile
 	avr/lib/avr5/at90scr100/Makefile
 	avr/lib/avr5/at90usb646/Makefile
 	avr/lib/avr5/at90usb647/Makefile
diff -Naurp ./devtools/gen-avr-lib-tree.sh ./devtools/gen-avr-lib-tree.sh
--- ./devtools/gen-avr-lib-tree.sh	2011-03-29 16:52:13.000000000 -0500
+++ ./devtools/gen-avr-lib-tree.sh	2011-03-29 16:54:43.000000000 -0500
@@ -159,6 +159,7 @@ at90can64:crtcan64.o:${DEV_DEFS}:${CFLAG
 at90can128:crtcan128.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
 at90pwm216:crt90pwm216.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
 at90pwm316:crt90pwm316.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
+at90pwm161:crt90pwm161.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
 at90scr100:crt90scr100.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
 at90usb646:crtusb646.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
 at90usb647:crtusb647.o:${DEV_DEFS}:${CFLAGS_SPACE}:${DEV_ASFLAGS};\
diff -Naurp ./doc/api/main_page.dox ./doc/api/main_page.dox
--- ./doc/api/main_page.dox	2011-03-29 16:52:13.000000000 -0500
+++ ./doc/api/main_page.dox	2011-03-29 16:54:43.000000000 -0500
@@ -230,6 +230,7 @@ compile-time.
 - at90pwm3
 - at90pwm3b
 - at90pwm316
+- at90pwm161
 - at90pwm81
 
 \par Smart Battery AVR Devices:
diff -Naurp ./doc/api/using-tools.dox ./doc/api/using-tools.dox
--- ./doc/api/using-tools.dox	2011-03-29 16:52:13.000000000 -0500
+++ ./doc/api/using-tools.dox	2011-03-29 16:54:43.000000000 -0500
@@ -290,6 +290,7 @@ AVR will be defined as well when using t
 
   <tr><td>avr5</td><td>at90can32</td><td>__AVR_AT90CAN32__</td></tr>
   <tr><td>avr5</td><td>at90can64</td><td>__AVR_AT90CAN64__</td></tr>
+  <tr><td>avr5</td><td>at90pwm161</td><td>__AVR_AT90PWM161__</td></tr>
   <tr><td>avr5</td><td>at90pwm216</td><td>__AVR_AT90PWM216__</td></tr>
   <tr><td>avr5</td><td>at90pwm316</td><td>__AVR_AT90PWM316__</td></tr>
   <tr><td>avr5</td><td>at90scr100</td><td>__AVR_AT90SCR100__</td></tr>
diff -Naurp ./include/avr/eeprom.h ./include/avr/eeprom.h
--- ./include/avr/eeprom.h	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/eeprom.h	2011-03-29 16:54:43.000000000 -0500
@@ -67,6 +67,8 @@
 # define _EEPROM_SUFFIX _90pwm216
 #elif defined (__AVR_AT90PWM316__)
 # define _EEPROM_SUFFIX _90pwm316
+#elif defined (__AVR_AT90PWM161__)
+# define _EEPROM_SUFFIX _90pwm161
 #elif defined (__AVR_AT90PWM81__)
 # define _EEPROM_SUFFIX _90pwm81
 #elif defined (__AVR_ATmega16M1__)
diff -Naurp ./include/avr/io90pwm161.h ./include/avr/io90pwm161.h
--- ./include/avr/io90pwm161.h	1969-12-31 18:00:00.000000000 -0600
+++ ./include/avr/io90pwm161.h	2011-03-29 16:56:13.000000000 -0500
@@ -0,0 +1 @@
+#define _VECTORS_SIZE 0
diff -Naurp ./include/avr/io.h ./include/avr/io.h
--- ./include/avr/io.h	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/io.h	2011-03-29 16:54:43.000000000 -0500
@@ -122,6 +122,8 @@
 #  include <avr/io90pwm216.h>
 #elif defined (__AVR_AT90PWM316__)
 #  include <avr/io90pwm316.h>
+#elif defined (__AVR_AT90PWM161__)
+#  include <avr/io90pwm161.h>
 #elif defined (__AVR_AT90PWM81__)
 #  include <avr/io90pwm81.h>
 #elif defined (__AVR_ATmega8U2__)
diff -Naurp ./include/avr/Makefile.am ./include/avr/Makefile.am
--- ./include/avr/Makefile.am	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/Makefile.am	2011-03-29 16:54:43.000000000 -0500
@@ -61,6 +61,7 @@ avr_HEADERS = \
     io90pwm216.h \
     io90pwm3b.h \
     io90pwm316.h \
+    io90pwm161.h \
     io90pwm81.h \
     io90scr100.h \
     ioa6289.h \
diff -Naurp ./include/avr/power.h ./include/avr/power.h
--- ./include/avr/power.h	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/power.h	2011-03-29 16:54:43.000000000 -0500
@@ -955,7 +955,8 @@ do{ \
 #define power_all_disable()     (PRR |= (uint8_t)((1<<PRADC)|(1<<PRSPI)|(1<<PRUSART)|(1<<PRTIM0)|(1<<PRTIM1)|(1<<PRPSC0)|(1<<PRPSC1)|(1<<PRPSC2)))
 
 
-#elif defined(__AVR_AT90PWM81__)
+#elif defined(__AVR_AT90PWM81__) \
+|| defined(__AVR_AT90PWM161__)
 
 #define power_adc_enable()      (PRR &= (uint8_t)~(1 << PRADC))
 #define power_adc_disable()     (PRR |= (uint8_t)(1 << PRADC))
@@ -1464,6 +1465,8 @@ do{ \
 || defined(__AVR_AT90PWM2B__) \
 || defined(__AVR_AT90PWM3__) \
 || defined(__AVR_AT90PWM3B__) \
+|| defined(__AVR_AT90PWM81__) \
+|| defined(__AVR_AT90PWM161__) \
 || defined(__AVR_AT90PWM216__) \
 || defined(__AVR_AT90PWM316__) \
 || defined(__AVR_AT90SCR100__) \
diff -Naurp ./include/avr/sleep.h ./include/avr/sleep.h
--- ./include/avr/sleep.h	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/sleep.h	2011-03-29 16:54:43.000000000 -0500
@@ -301,6 +301,7 @@
 
 #elif defined(__AVR_AT90PWM216__) \
 || defined(__AVR_AT90PWM316__) \
+|| defined(__AVR_AT90PWM161__) \
 || defined(__AVR_AT90PWM81__)
 
     #define SLEEP_MODE_IDLE         0
diff -Naurp ./include/avr/wdt.h ./include/avr/wdt.h
--- ./include/avr/wdt.h	2011-03-29 16:52:13.000000000 -0500
+++ ./include/avr/wdt.h	2011-03-29 16:54:43.000000000 -0500
@@ -180,6 +180,7 @@ __asm__ __volatile__ ( \
 || defined(__AVR_AT90PWM3__) \
 || defined(__AVR_AT90PWM316__) \
 || defined(__AVR_AT90PWM3B__) \
+|| defined(__AVR_AT90PWM161__) \
 || defined(__AVR_AT90PWM81__) \
 || defined(__AVR_AT90USB1286__) \
 || defined(__AVR_AT90USB1287__) \
@@ -419,7 +420,7 @@ __asm__ __volatile__ (  \
     ATmega8HVA, ATmega16HVA, ATmega32HVB,
     ATmega406, ATmega1284P,
     AT90PWM1, AT90PWM2, AT90PWM2B, AT90PWM3, AT90PWM3B, AT90PWM216, AT90PWM316,
-    AT90PWM81,
+    AT90PWM81, AT90PWM161,
     AT90USB82, AT90USB162,
     AT90USB646, AT90USB647, AT90USB1286, AT90USB1287,
     ATtiny48, ATtiny88.
@@ -440,7 +441,7 @@ __asm__ __volatile__ (  \
     ATmega8HVA, ATmega16HVA, ATmega32HVB,
     ATmega406, ATmega1284P,
     AT90PWM1, AT90PWM2, AT90PWM2B, AT90PWM3, AT90PWM3B, AT90PWM216, AT90PWM316,
-    AT90PWM81,
+    AT90PWM81, AT90PWM161,
     AT90USB82, AT90USB162,
     AT90USB646, AT90USB647, AT90USB1286, AT90USB1287,
     ATtiny48, ATtiny88.
